modules/CmpdRegBulkLoader/src/server/routes/CmpdRegBulkLoaderRoutes.coffee

```diff
@@ -171,32 +182,74 @@ exports.validateCmpds = (req, resp) ->
 	exports.registerCmpds(req, resp)
 
 exports.registerCmpds = (req, resp) ->
+	serverUtilityFunctions = require './ServerUtilityFunctions.js'
+	config = require '../conf/compiled/conf.js'
 	req.connection.setTimeout 6000000
-	createSummaryZip = (fileName, originalFileName, json) ->
-		#remove .sdf from fileName and originalFileName
-		fileName = fileName.substring(0, fileName.length-4)
-		originalFileName = originalFileName.substring(0, originalFileName.length-4)
-		zipFileName = fileName+".zip"
-		zipFileDisplayName = originalFileName+".zip"
-		fs = require 'fs'
-		JSZip = require 'jszip'
-		zip = new JSZip()
 
+	createSummaryZip = (fileName, originalFileName, json) ->
+		# This is passed as just a name but it exists in the bulk_load_sub_folder
+		originalFilePath = path.join(exports.BULKLOAD_SUB_FOLDER, fileName)
+
+		# Replace the extension with .zip to create the zip file name
+		zipFileName =  path.basename(fileName, path.extname(fileName)) + '.zip'
+		zipFileDisplayName = path.basename(originalFileName, path.extname(originalFileName)) + '.zip'
+		
+		# Create the zip file
 		for rFile in json.reportFiles
-			serverUtilityFunctions = require './ServerUtilityFunctions.js'
-			config = require '../conf/compiled/conf.js'
-			splitNames = rFile.split (path.sep+"cmpdreg_bulkload"+path.sep)
+			splitNames = rFile.split (path.sep+exports.BULKLOAD_SUB_FOLDER+path.sep)
 			rFileName = splitNames[1]
 			rFileName = rFileName.replace(path.sep, '');
 			# rename to use the original name rather than the actual on disk name
 			rFileName = rFileName.replace(fileName, originalFileName)
 			zip.file(rFileName, fs.readFileSync(rFile))
+
 		origUploadsPath = serverUtilityFunctions.makeAbsolutePath config.all.server.datafiles.relative_path
-		zipFilePath = origUploadsPath + "cmpdreg_bulkload" + path.sep + zipFileName
-		console.log zipFilePath
+		shortZipName = path.join(exports.BULKLOAD_SUB_FOLDER, zipFileName)
+		zipFilePath = path.join(origUploadsPath, shortZipName)
+
+		console.log "Creating zip file #{zipFilePath}"
 		fstream = zip.generateNodeStream({type:"nodebuffer", streamFiles:true}).pipe(fs.createWriteStream(zipFilePath))
+		
 		fstream.on 'finish', ->
-			console.log "finished create write stream"
+			console.log "Zip file #{zipFilePath} created"
```

Comment:
This is all the new logic for moving the files to a subfolder. I should probably move this to it's own method.